test/e2e/tas/node_failure_test.go

```diff
@@ -0,0 +1,233 @@
+				Image(util.E2eTestAgnHostImage, util.BehaviorWaitForDeletion).
+				Obj()
+			util.MustCreate(ctx, k8sClient, sampleJob)
+
+			ginkgo.By("Job is unsuspended, and has all Pods active and ready", func() {
+				gomega.Eventually(func(g gomega.Gomega) {
+					g.Expect(k8sClient.Get(ctx, client.ObjectKeyFromObject(sampleJob), sampleJob)).To(gomega.Succeed())
+					g.Expect(sampleJob.Spec.Suspend).Should(gomega.Equal(ptr.To(false)))
+				}, util.LongTimeout, util.Interval).Should(gomega.Succeed())
+				gomega.Eventually(func(g gomega.Gomega) {
+					g.Expect(k8sClient.Get(ctx, client.ObjectKeyFromObject(sampleJob), sampleJob)).To(gomega.Succeed())
+					g.Expect(sampleJob.Status.Active).Should(gomega.Equal(int32(numPods)))
+				}, util.LongTimeout, util.Interval).Should(gomega.Succeed())
+				gomega.Eventually(func(g gomega.Gomega) {
+					g.Expect(k8sClient.Get(ctx, client.ObjectKeyFromObject(sampleJob), sampleJob)).To(gomega.Succeed())
+					g.Expect(sampleJob.Status.Ready).Should(gomega.Equal(ptr.To(int32(numPods))))
+				}, util.LongTimeout, util.Interval).Should(gomega.Succeed())
+			})
+		})
+		ginkgo.AfterEach(func() {
+			gomega.Expect(util.DeleteAllJobsInNamespace(ctx, k8sClient, ns)).Should(gomega.Succeed())
+			gomega.Expect(util.DeleteWorkloadsInNamespace(ctx, k8sClient, ns)).Should(gomega.Succeed())
+			util.ExpectObjectToBeDeleted(ctx, k8sClient, localQueue, true)
+			util.ExpectObjectToBeDeleted(ctx, k8sClient, clusterQueue, true)
+			util.ExpectObjectToBeDeleted(ctx, k8sClient, tasFlavor, true)
+			util.ExpectObjectToBeDeleted(ctx, k8sClient, topology, true)
+			util.ExpectAllPodsInNamespaceDeleted(ctx, k8sClient, ns)
+		})
+
+		ginkgo.It("Should update nodesToReplace at the workload when a node fails", func() {
+			pods := &corev1.PodList{}
+			ginkgo.By("Ensure all pods are running on nodes", func() {
+				listOpts := &client.ListOptions{
+					FieldSelector: fields.OneTermNotEqualSelector("spec.nodeName", ""),
+				}
+				gomega.Eventually(func(g gomega.Gomega) {
+					g.Expect(k8sClient.List(ctx, pods, client.InNamespace(ns.Name), listOpts)).To(gomega.Succeed())
+					g.Expect(pods.Items).Should(gomega.HaveLen(numPods))
+				}, util.Timeout, util.Interval).Should(gomega.Succeed())
+			})
+			node := &corev1.Node{}
+			chosenPod := pods.Items[3]
+			var wlName string
+			ginkgo.By("Get node running the chosen pod", func() {
+				var found bool
+				wlName, found = chosenPod.Annotations[kueuealpha.WorkloadAnnotation]
+				gomega.Expect(found).Should(gomega.BeTrue())
+				nodeList := &corev1.NodeList{}
+				gomega.Eventually(func(g gomega.Gomega) {
+					g.Expect(k8sClient.List(ctx, nodeList)).To(gomega.Succeed())
+				}, util.ShortTimeout, util.Interval).Should(gomega.Succeed())
+				for _, n := range nodeList.Items {
+					if n.Name == chosenPod.Spec.NodeName {
+						node = &n
+						break
+					}
+				}
+			})
+
+			// Store original node condition to restore it later
+			defer func() {
```

Comment:
Use `ginkgo.DeferCleanup`. `defer func` runs when exiting the test scenario.

It does not mostly matter, but there might be cases when the ginkgo functions create just execution plan instead of executing immediately, for example when using parallel execution. So, it is better to give full control on the order of execution to ginkgo.